Jamfile

```diff
@@ -511,10 +553,20 @@ feature.compose <encryption>off : <define>TORRENT_DISABLE_ENCRYPTION ;
 feature mutable-torrents : on off : composite propagated link-incompatible ;
 feature.compose <mutable-torrents>off : <define>TORRENT_DISABLE_MUTABLE_TORRENTS ;
 
-feature crypto : built-in openssl libcrypto gcrypt : composite propagated ;
+feature crypto : built-in openssl libcrypto gcrypt wolfssl : composite propagated ;
 feature.compose <crypto>openssl : <define>TORRENT_USE_LIBCRYPTO <define>TORRENT_USE_OPENSSL <define>OPENSSL_NO_SSL2 ;
 feature.compose <crypto>libcrypto : <define>TORRENT_USE_LIBCRYPTO ;
 feature.compose <crypto>gcrypt : <define>TORRENT_USE_LIBGCRYPT ;
+feature.compose <crypto>wolfssl
+	: <define>TORRENT_USE_WOLFSSL
+	<define>TORRENT_USE_LIBCRYPTO
+	<define>TORRENT_USE_OPENSSL
```

Comment:
so, wolfSSL provides openssl headers? doesn't that conflict with openssl? can these libraries not be installed at the same time?
Do you think it would be impractical to, instead of defining both `TORRENT_USE_OPENSSL` and `TORRENT_USE_WOLFSSL`, change the code to check for both macros? part of me feels like that would be a bit cleaner, just in case there are deviations at some places.

Reply:
well, they only achieved that compatibility with `asio` (and others) providing the emulation layer, so for this case, the openssl headers will (and need) to be there. I also think that checking for both macros is very aggressive on the code, and we can always check for `TORRENT_USE_WOLFSSL` for special cases.

Reply:
I see. and so far there is not a single need to check `TORRENT_USE_WOLFSSL`, is that right? If the openssl emulation is accurate enough to "just work", we should probably not even define `TORRENT_USE_WOLFSSL` either, right?

Reply:
right, except that I assume `define TORRENT_CRYPTO_NAMESPACE inline namespace wcrypto {` should be accurate (or clear) for clients, or not?